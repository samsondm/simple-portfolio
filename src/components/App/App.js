import React, { Component } from 'react';
import {
  Route,
  Switch,
  // withRouter,
  Link
} from 'react-router-dom';
import './App.css';
import Welcome from '../Welcome/Welcome';
import Projects from '../Projects/Projects';
import Carousel from '../Carousel/Carousel';
import { calc, drum, pomodoro, random, weather } from '../../images';
import CalcContainer from '../Calculator/CalcContainer';
import DrumMachine from '../DrumMachine/DrumMachine';
import Layer from '../Layer/Layer';
import PomodoroClock from '../PomodoroClock/PomodoroClock';
import RandomQuoteMachine from '../RandomQuoteMachine/RandomQuoteMachine';
import WeatherApp from '../WeatherApp/WeatherApp';
import ProjectWrapper from '../ProjectWrapper/ProjectWrapper';
import scrollIntoView from '../../utility/scrollIntoView';
import ProjectSlide from '../ProjectSlide/ProjectSlide';

class App extends Component {
  componentsMap = {
    'random-quote-machine': {
      component: RandomQuoteMachine,
      title: 'Random Quote Machine',
      image: random,
      description: 'Get random quotes about design.',
      stack: 'React, Redux, Redux-Thunk, REST API'
    },
    'drum-machine': {
      component: DrumMachine,
      title: 'Drum Machine',
      image: drum,
      description: 'Drop some beats.',
      stack: 'React, Redux, SASS'
    },
    'calc': {
      component: CalcContainer,
      title: 'Calculator',
      image: calc,
      description: 'Simple javascript calculator.',
      stack: 'React, Redux, Redux-Thunk'
    },
    'pomodoro-clock': {
      component: PomodoroClock,
      title: 'Pomodoro Clock',
      image: pomodoro,
      description: 'Use the Pomodoro Technique to help you increase your focus.',
      stack: 'React, SASS'
    },
    'weather-app': {
      component: WeatherApp,
      title: 'Weather App',
      image: weather,
      description: 'Check weather in your current location or in other cities.',
      stack: 'React, SASS, REST API'
    }
  }

  projectsRef = React.createRef();
  scrollToProjects = () => {
    scrollIntoView(this.projectsRef.current);
  }

  // handle hover
  container = document.body;
  hasHoverClass = false;
  lastTouchTime = 0;

  // enable hover affect on mousemove and touchstart events
  enableHover = (e) => {
    if (this.hasHoverClass) return;
    // filter touch emulated mousemove events that are trigered after touchend event
    const filterDelta = 650;
    if (e.type !== 'touchstart' && performance.now() - this.lastTouchTime < filterDelta) return;

    this.container.classList.add('hasHover');
    this.hasHoverClass = true;
  };

  // disable hover affect on touchend
  disableHover = (e) => {
    if (!this.hasHoverClass) return;

    this.container.classList.remove('hasHover');
    this.hasHoverClass = false;
  };

  // save time of the last touchend event
  updateLastTouchTime = () => {
    this.lastTouchTime = performance.now();
  };

  componentDidMount() {
    document.addEventListener('touchend', this.updateLastTouchTime, true);
    document.addEventListener('touchend', this.disableHover, true);
    document.addEventListener('touchstart', this.enableHover, true);
    document.addEventListener('mousemove', this.enableHover, true);
  }

  componentWillUnmount() {
    document.removeEventListener('touchend', this.updateLastTouchTime, true);
    document.removeEventListener('touchend', this.disableHover, true);
    document.removeEventListener('touchstart', this.enableHover, true);
    document.removeEventListener('mousemove', this.enableHover, true);

    this.disableHover();
  }


  render() {
    const projectSlideArray = Object.keys(this.componentsMap).map(value =>
      <ProjectSlide
        imgUrl={this.componentsMap[value].image}
        title={this.componentsMap[value].title}
        description={this.componentsMap[value].description}
        stack={this.componentsMap[value].stack}
        key={value}
      >
        <Link to={"/" + value} tabIndex="-1" children="explore" />
      </ProjectSlide>
    );

    return (
      <div className="App" >
        <Welcome onClick={this.scrollToProjects} />
        <Projects ref={this.projectsRef}>
          <Carousel >
            {projectSlideArray}
          </Carousel>
        </Projects>

        <Switch>
          <Route path="/layer" component={Layer} />
          <Route
            path="/:project"
            render={
              ({ history, match }) => {
                const projectName = match.params.project,
                  Project = this.componentsMap[projectName].component,
                  title = this.componentsMap[projectName].title;
                return (
                  <Layer history={history}>
                    <ProjectWrapper title={title} className={`${projectName}-wrapper`} history={history}>
                      <Project />
                    </ProjectWrapper>
                  </Layer>
                );
              }
            }
          />
        </Switch>
      </div>
    );
  }
}

export default App;