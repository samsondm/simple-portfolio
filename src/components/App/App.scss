.App {
  width: 100%;
  font-family: 'Roboto', sans-serif;
  background: linear-gradient(#3f87a6, #ebf8e1 100vh, #f69d3c);
}

// fix for ie
.min-height-flex-fix {
  &:after {
    content: '';
    display: block; // changed from table because of firefox bug with pseudo-elements
    min-height: inherit;
  }
}

.icon-vertical-center {
  vertical-align: middle;
}