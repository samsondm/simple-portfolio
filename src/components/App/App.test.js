import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

const withMemoryRouter = (Component) => <MemoryRouter><Component /></MemoryRouter>;

describe('App', () => {
  it('renders without crashing', () => {
    mount(withMemoryRouter(App));
  });

  it('snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});