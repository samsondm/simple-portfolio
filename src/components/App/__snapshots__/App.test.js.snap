// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Welcome
          onClick={[Function]}
        />,
        <ForwardRef>
          <Carousel>
            <ProjectSlide
              description="Get random quotes about design."
              imgUrl="random-quote-machine.png"
              stack="React, Redux, Redux-Thunk, REST API"
              title="Random Quote Machine"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/random-quote-machine"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Drop some beats."
              imgUrl="drum-machine.png"
              stack="React, Redux, SASS"
              title="Drum Machine"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/drum-machine"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Simple javascript calculator."
              imgUrl="calc.png"
              stack="React, Redux, Redux-Thunk"
              title="Calculator"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/calc"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Use the Pomodoro Technique to help you increase your focus."
              imgUrl="pomodoro-clock.png"
              stack="React, SASS"
              title="Pomodoro Clock"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/pomodoro-clock"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Check weather in your current location or in other cities."
              imgUrl="weather-app.png"
              stack="React, SASS, REST API"
              title="Weather App"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/weather-app"
              >
                explore
              </Link>
            </ProjectSlide>
          </Carousel>
        </ForwardRef>,
        <Switch>
          <Route
            component={[Function]}
            path="/layer"
          />
          <Route
            path="/:project"
            render={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Carousel>
            <ProjectSlide
              description="Get random quotes about design."
              imgUrl="random-quote-machine.png"
              stack="React, Redux, Redux-Thunk, REST API"
              title="Random Quote Machine"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/random-quote-machine"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Drop some beats."
              imgUrl="drum-machine.png"
              stack="React, Redux, SASS"
              title="Drum Machine"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/drum-machine"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Simple javascript calculator."
              imgUrl="calc.png"
              stack="React, Redux, Redux-Thunk"
              title="Calculator"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/calc"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Use the Pomodoro Technique to help you increase your focus."
              imgUrl="pomodoro-clock.png"
              stack="React, SASS"
              title="Pomodoro Clock"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/pomodoro-clock"
              >
                explore
              </Link>
            </ProjectSlide>
            <ProjectSlide
              description="Check weather in your current location or in other cities."
              imgUrl="weather-app.png"
              stack="React, SASS, REST API"
              title="Weather App"
            >
              <Link
                replace={false}
                tabIndex="-1"
                to="/weather-app"
              >
                explore
              </Link>
            </ProjectSlide>
          </Carousel>,
        },
        "ref": Object {
          "current": null,
        },
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ProjectSlide
                description="Get random quotes about design."
                imgUrl="random-quote-machine.png"
                stack="React, Redux, Redux-Thunk, REST API"
                title="Random Quote Machine"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/random-quote-machine"
                >
                  explore
                </Link>
              </ProjectSlide>,
              <ProjectSlide
                description="Drop some beats."
                imgUrl="drum-machine.png"
                stack="React, Redux, SASS"
                title="Drum Machine"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/drum-machine"
                >
                  explore
                </Link>
              </ProjectSlide>,
              <ProjectSlide
                description="Simple javascript calculator."
                imgUrl="calc.png"
                stack="React, Redux, Redux-Thunk"
                title="Calculator"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/calc"
                >
                  explore
                </Link>
              </ProjectSlide>,
              <ProjectSlide
                description="Use the Pomodoro Technique to help you increase your focus."
                imgUrl="pomodoro-clock.png"
                stack="React, SASS"
                title="Pomodoro Clock"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/pomodoro-clock"
                >
                  explore
                </Link>
              </ProjectSlide>,
              <ProjectSlide
                description="Check weather in your current location or in other cities."
                imgUrl="weather-app.png"
                stack="React, SASS, REST API"
                title="Weather App"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/weather-app"
                >
                  explore
                </Link>
              </ProjectSlide>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "random-quote-machine",
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  tabIndex="-1"
                  to="/random-quote-machine"
                >
                  explore
                </Link>,
                "description": "Get random quotes about design.",
                "imgUrl": "random-quote-machine.png",
                "stack": "React, Redux, Redux-Thunk, REST API",
                "title": "Random Quote Machine",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "explore",
                  "replace": false,
                  "tabIndex": "-1",
                  "to": "/random-quote-machine",
                },
                "ref": null,
                "rendered": "explore",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "drum-machine",
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  tabIndex="-1"
                  to="/drum-machine"
                >
                  explore
                </Link>,
                "description": "Drop some beats.",
                "imgUrl": "drum-machine.png",
                "stack": "React, Redux, SASS",
                "title": "Drum Machine",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "explore",
                  "replace": false,
                  "tabIndex": "-1",
                  "to": "/drum-machine",
                },
                "ref": null,
                "rendered": "explore",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "calc",
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  tabIndex="-1"
                  to="/calc"
                >
                  explore
                </Link>,
                "description": "Simple javascript calculator.",
                "imgUrl": "calc.png",
                "stack": "React, Redux, Redux-Thunk",
                "title": "Calculator",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "explore",
                  "replace": false,
                  "tabIndex": "-1",
                  "to": "/calc",
                },
                "ref": null,
                "rendered": "explore",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "pomodoro-clock",
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  tabIndex="-1"
                  to="/pomodoro-clock"
                >
                  explore
                </Link>,
                "description": "Use the Pomodoro Technique to help you increase your focus.",
                "imgUrl": "pomodoro-clock.png",
                "stack": "React, SASS",
                "title": "Pomodoro Clock",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "explore",
                  "replace": false,
                  "tabIndex": "-1",
                  "to": "/pomodoro-clock",
                },
                "ref": null,
                "rendered": "explore",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "weather-app",
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  tabIndex="-1"
                  to="/weather-app"
                >
                  explore
                </Link>,
                "description": "Check weather in your current location or in other cities.",
                "imgUrl": "weather-app.png",
                "stack": "React, SASS, REST API",
                "title": "Weather App",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "explore",
                  "replace": false,
                  "tabIndex": "-1",
                  "to": "/weather-app",
                },
                "ref": null,
                "rendered": "explore",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/layer"
            />,
            <Route
              path="/:project"
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/layer",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/:project",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Welcome
            onClick={[Function]}
          />,
          <ForwardRef>
            <Carousel>
              <ProjectSlide
                description="Get random quotes about design."
                imgUrl="random-quote-machine.png"
                stack="React, Redux, Redux-Thunk, REST API"
                title="Random Quote Machine"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/random-quote-machine"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Drop some beats."
                imgUrl="drum-machine.png"
                stack="React, Redux, SASS"
                title="Drum Machine"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/drum-machine"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Simple javascript calculator."
                imgUrl="calc.png"
                stack="React, Redux, Redux-Thunk"
                title="Calculator"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/calc"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Use the Pomodoro Technique to help you increase your focus."
                imgUrl="pomodoro-clock.png"
                stack="React, SASS"
                title="Pomodoro Clock"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/pomodoro-clock"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Check weather in your current location or in other cities."
                imgUrl="weather-app.png"
                stack="React, SASS, REST API"
                title="Weather App"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/weather-app"
                >
                  explore
                </Link>
              </ProjectSlide>
            </Carousel>
          </ForwardRef>,
          <Switch>
            <Route
              component={[Function]}
              path="/layer"
            />
            <Route
              path="/:project"
              render={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Carousel>
              <ProjectSlide
                description="Get random quotes about design."
                imgUrl="random-quote-machine.png"
                stack="React, Redux, Redux-Thunk, REST API"
                title="Random Quote Machine"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/random-quote-machine"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Drop some beats."
                imgUrl="drum-machine.png"
                stack="React, Redux, SASS"
                title="Drum Machine"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/drum-machine"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Simple javascript calculator."
                imgUrl="calc.png"
                stack="React, Redux, Redux-Thunk"
                title="Calculator"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/calc"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Use the Pomodoro Technique to help you increase your focus."
                imgUrl="pomodoro-clock.png"
                stack="React, SASS"
                title="Pomodoro Clock"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/pomodoro-clock"
                >
                  explore
                </Link>
              </ProjectSlide>
              <ProjectSlide
                description="Check weather in your current location or in other cities."
                imgUrl="weather-app.png"
                stack="React, SASS, REST API"
                title="Weather App"
              >
                <Link
                  replace={false}
                  tabIndex="-1"
                  to="/weather-app"
                >
                  explore
                </Link>
              </ProjectSlide>
            </Carousel>,
          },
          "ref": Object {
            "current": null,
          },
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <ProjectSlide
                  description="Get random quotes about design."
                  imgUrl="random-quote-machine.png"
                  stack="React, Redux, Redux-Thunk, REST API"
                  title="Random Quote Machine"
                >
                  <Link
                    replace={false}
                    tabIndex="-1"
                    to="/random-quote-machine"
                  >
                    explore
                  </Link>
                </ProjectSlide>,
                <ProjectSlide
                  description="Drop some beats."
                  imgUrl="drum-machine.png"
                  stack="React, Redux, SASS"
                  title="Drum Machine"
                >
                  <Link
                    replace={false}
                    tabIndex="-1"
                    to="/drum-machine"
                  >
                    explore
                  </Link>
                </ProjectSlide>,
                <ProjectSlide
                  description="Simple javascript calculator."
                  imgUrl="calc.png"
                  stack="React, Redux, Redux-Thunk"
                  title="Calculator"
                >
                  <Link
                    replace={false}
                    tabIndex="-1"
                    to="/calc"
                  >
                    explore
                  </Link>
                </ProjectSlide>,
                <ProjectSlide
                  description="Use the Pomodoro Technique to help you increase your focus."
                  imgUrl="pomodoro-clock.png"
                  stack="React, SASS"
                  title="Pomodoro Clock"
                >
                  <Link
                    replace={false}
                    tabIndex="-1"
                    to="/pomodoro-clock"
                  >
                    explore
                  </Link>
                </ProjectSlide>,
                <ProjectSlide
                  description="Check weather in your current location or in other cities."
                  imgUrl="weather-app.png"
                  stack="React, SASS, REST API"
                  title="Weather App"
                >
                  <Link
                    replace={false}
                    tabIndex="-1"
                    to="/weather-app"
                  >
                    explore
                  </Link>
                </ProjectSlide>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "random-quote-machine",
                "nodeType": "function",
                "props": Object {
                  "children": <Link
                    replace={false}
                    tabIndex="-1"
                    to="/random-quote-machine"
                  >
                    explore
                  </Link>,
                  "description": "Get random quotes about design.",
                  "imgUrl": "random-quote-machine.png",
                  "stack": "React, Redux, Redux-Thunk, REST API",
                  "title": "Random Quote Machine",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "explore",
                    "replace": false,
                    "tabIndex": "-1",
                    "to": "/random-quote-machine",
                  },
                  "ref": null,
                  "rendered": "explore",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "drum-machine",
                "nodeType": "function",
                "props": Object {
                  "children": <Link
                    replace={false}
                    tabIndex="-1"
                    to="/drum-machine"
                  >
                    explore
                  </Link>,
                  "description": "Drop some beats.",
                  "imgUrl": "drum-machine.png",
                  "stack": "React, Redux, SASS",
                  "title": "Drum Machine",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "explore",
                    "replace": false,
                    "tabIndex": "-1",
                    "to": "/drum-machine",
                  },
                  "ref": null,
                  "rendered": "explore",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "calc",
                "nodeType": "function",
                "props": Object {
                  "children": <Link
                    replace={false}
                    tabIndex="-1"
                    to="/calc"
                  >
                    explore
                  </Link>,
                  "description": "Simple javascript calculator.",
                  "imgUrl": "calc.png",
                  "stack": "React, Redux, Redux-Thunk",
                  "title": "Calculator",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "explore",
                    "replace": false,
                    "tabIndex": "-1",
                    "to": "/calc",
                  },
                  "ref": null,
                  "rendered": "explore",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "pomodoro-clock",
                "nodeType": "function",
                "props": Object {
                  "children": <Link
                    replace={false}
                    tabIndex="-1"
                    to="/pomodoro-clock"
                  >
                    explore
                  </Link>,
                  "description": "Use the Pomodoro Technique to help you increase your focus.",
                  "imgUrl": "pomodoro-clock.png",
                  "stack": "React, SASS",
                  "title": "Pomodoro Clock",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "explore",
                    "replace": false,
                    "tabIndex": "-1",
                    "to": "/pomodoro-clock",
                  },
                  "ref": null,
                  "rendered": "explore",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "weather-app",
                "nodeType": "function",
                "props": Object {
                  "children": <Link
                    replace={false}
                    tabIndex="-1"
                    to="/weather-app"
                  >
                    explore
                  </Link>,
                  "description": "Check weather in your current location or in other cities.",
                  "imgUrl": "weather-app.png",
                  "stack": "React, SASS, REST API",
                  "title": "Weather App",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "explore",
                    "replace": false,
                    "tabIndex": "-1",
                    "to": "/weather-app",
                  },
                  "ref": null,
                  "rendered": "explore",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/layer"
              />,
              <Route
                path="/:project"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/layer",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/:project",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
